packages/openapi-generator/src/openapi-types.ts

```diff
@@ -14,24 +15,9 @@ export interface OpenApiDocument {
   serviceName: string;
 
   /**
-   * Name of the npm package, if there is a package.json.
+   * Configuration as defined in the per service configuration.
    */
-  npmPackageName: string;
-
-  /**
-   * Name of the directory where the service will be generated to.
-   */
-  directoryName: string;
-
-  /**
-   * Name of the original specification file.
-   */
-  originalFileName: string;
-
-  /**
-   * Path of the original specification file.
-   */
-  filePath: string;
+  serviceConfig: ServiceConfig;
 
```

Comment:
Question: 4 properties are deleted here, but the `ServiceConfig` only contains `packageName` and `directoryName`. Are the missing two properties not useful?
I know we can get e.g., `filePath`, but if we dont need to reuse it, I'm happy to delete.

Reply:
It is now passed as a separate parameters. I was considering this for the serviceConfig as well, but then decided against it to be pragmatic.